### ISSUE TITLE
Doesn't replace the whole phrase or file name.

### ISSUE THREAD

**Issue by dummifiedme** [tadashi-aikawa/obsidian-various-complements-plugin#58]
I have been using the plugin for a while now, and I noticed that many times when we want to type out a phrase or a file name with multiple words, it suggests at the second or even third word, and when we select it, it just replaces that particular word with the hole suggestion. 

**Lets take an example first:**

If I have a file name `[[ Civil Society Board ]]` and while writing a text in another note I go like...

```
We are working on getting accredited by Civil Society Bo

```

At that point, it will suggest me `[[ Civil Society Board ]]` and I make the selection.

The text now reads:

```
We are working on getting accredited by Civil Society [[ Civil Society Board ]]

```

This has been happening a lot lately. It should ideally replace all three sentences or atleast suggest something on the first word. I have tried it with having a list as large as 6.

- - - 
I know this might not be easy, but if you could look at it, it might be better experience for everyone :)

**Comment by tadashi-aikawa** [comment]
Hi, @dummifiedme. Thank you for your report :)

First, it is not a bug, but as you said, I think it is more convenient and intuitive.

If you want to resolve this issue, I recommend you to set `prefix` to a `Match strategy` option.

![image](https://user-images.githubusercontent.com/9500018/147410045-18fe9c9c-0460-4bc0-a6ad-b3749bd4e6cc.png)

If I understand you correctly, do you want to set `partial` to a `Match strategy` option?

**Comment by dummifiedme** [comment]
> First, it is not a bug, but as you said, I think it is more convenient and intuitive.

Yeah, removed the [Bug] mark. Hehe! 

Yes, I was using Partial for some days. Should I stick to prefix?

**Comment by tadashi-aikawa** [comment]
Thanks!

> Should I stick to prefix?

You don't need to stick to `Prefix` if you want to use `Partial`. But if not, it resolves the problem because it doesn't suggest `[[ Civil Society Board ]]` from `Bo`.

If you intentionally use `Partial`, I'll challenge to implement it.

**Comment by dummifiedme** [comment]
> If you intentionally use Partial, I'll challenge to implement it.

A challenge for me or yourself? :no_mouth:

**Comment by tadashi-aikawa** [comment]
> A challenge for me or yourself? 😶

Both. I've been thinking about this for a while but deprioritized. Because I had felt many users who use English don't need it. 

🤔 oO (They use `Prefix` not `Partial` )

**Comment by dummifiedme** [comment]
If a user uses it to quickly grab internal links  then partial is better since the file names doesnt really need to be grammatically in order. If we could match partials clubbed in two or three words, we might find the note already in the  system. Even better, if we could assign a toggle to have a shortcut to choose and replace the filename with the text we just wrote. 

eg. If we have a sentence going like `We should really evaluate the role of procrastination` and if there is a note called `Procrastination - Role`, it could replace the text with a link and alias,  or even if the note is `Role of Procrastination`, there could be 100s of notes starting `role` and it might not show this one on the top since we are only using Prefix!


Did I make it clear or should I try to explain again? is there something I am missing?

**Comment by tadashi-aikawa** [comment]
@dummifiedme 
Thank you for your kind explanation! I agree with you 👍 . (Since I almost write not English but Japanese in Obsidian)

**Comment by tadashi-aikawa** [comment]
@dummifiedme 
I have updated beta version to [obsidian-various-complements-plugin-beta](https://github.com/tadashi-aikawa/obsidian-various-complements-plugin-beta/releases/tag/4.0.0.beta2) 🚀 

If you use [BRAT](https://github.com/TfTHacker/obsidian42-brat), please confirm whether it works as you expected. If you don't use it, please verify the below GIF.

![2021-12-30](https://user-images.githubusercontent.com/9500018/147746521-11912fa9-7ea8-4982-a0cc-06ddd98fd0a1.gif)

**Comment by dummifiedme** [comment]
Seems great. Good work!!


- Does it take two words at a time or more?
- And if the words match perfectly, would it move the suggestion to the top, or the later (single) word will always be at the top?

**Comment by tadashi-aikawa** [comment]
> Does it take two words at a time or more?

Yes.

![2021-12-30](https://user-images.githubusercontent.com/9500018/147749539-ed05aa76-075c-45f7-897a-eca9af247053.gif)
 
> if the words match perfectly, would it move the suggestion to the top, or the later (single) word will always be at the top?

No. `the later (single) word will always be at the top` is right.

That isn't easy with a kept performance good. https://github.com/tadashi-aikawa/obsidian-various-complements-plugin/issues/24 is a related issue.

**Comment by dummifiedme** [comment]
> That isn't easy with a kept performance good. 

Then its fine! 
Seems good to go!!

**Comment by tadashi-aikawa** [comment]
I want to release v4.0.0 this weekend. please wait. 👍

**Comment by dummifiedme** [comment]
Sure! Should I close this now?

**Comment by tadashi-aikawa** [comment]
No. After the release and your confirmation :)

**Comment by tadashi-aikawa** [comment]
I have updated beta version to [obsidian-various-complements-plugin-beta 4.0.0.beta5](https://github.com/tadashi-aikawa/obsidian-various-complements-plugin-beta/releases/tag/4.0.0.beta5) 🚀

**Comment by tadashi-aikawa** [comment]
@dummifiedme 
I have released to [v4.0.0](https://github.com/tadashi-aikawa/obsidian-various-complements-plugin/releases/tag/4.0.0) 🚀

Please confirm whether it works as you expected.

**Comment by dummifiedme** [comment]
I would have loved if it had the perfect matches on top with partial, but still I am happy with it. 
It works fine :)
Thanks a lot!!

**Comment by tadashi-aikawa** [comment]
@dummifiedme 
Then, please watch #24 👍

**Comment by tadashi-aikawa** [comment]
@dummifiedme 
I removed the `Overwrite duplicated existed phrase` option because I have implemented #24 😄 

https://github.com/tadashi-aikawa/obsidian-various-complements-plugin/issues/24#issuecomment-1013699255

I don't have released it as an official release yet. So please wait for v5.0.0 if you don't use BRAT :)

**Comment by dummifiedme** [comment]
> @dummifiedme I removed the `Overwrite duplicated existed phrase` option because I have implemented #24 😄
> 
> [#24 (comment)](https://github.com/tadashi-aikawa/obsidian-various-complements-plugin/issues/24#issuecomment-1013699255)
> 
> I don't have released it as an official release yet. So please wait for v5.0.0 if you don't use BRAT :)

yes, I am watching every activity on thi plugin 🤪
